collect_app/src/main/java/org/odk/collect/android/fragments/AppListFragment.java

```diff
@@ -0,0 +1,272 @@
+package org.odk.collect.android.fragments;
+
+import android.content.res.Configuration;
+import android.database.Cursor;
+import android.os.Bundle;
+import android.support.v4.app.ListFragment;
+import android.support.v4.widget.DrawerLayout;
+import android.support.v7.app.ActionBarDrawerToggle;
+import android.view.Gravity;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.database.ActivityLogger;
+import org.odk.collect.android.provider.InstanceProviderAPI;
+
+import java.util.List;
+
+
+abstract class AppListFragment extends ListFragment {
+    public static final int MENU_SORT = Menu.FIRST;
+    protected final ActivityLogger logger = Collect.getInstance().getActivityLogger();
+    protected String[] mSortingOptions;
+    View rootView;
+    private ListView mDrawerList;
+    private DrawerLayout mDrawerLayout;
+    private ActionBarDrawerToggle mDrawerToggle;
+
+    // toggles to all checked or all unchecked
+    // returns:
+    // true if result is all checked
+    // false if result is all unchecked
+    //
+    // Toggle behavior is as follows:
+    // if ANY items are unchecked, check them all
+    // if ALL items are checked, uncheck them all
+    public static boolean toggleChecked(ListView lv) {
+        // shortcut null case
+        if (lv == null) return false;
+
+        boolean newCheckState = lv.getCount() > lv.getCheckedItemCount();
+        setAllToCheckedState(lv, newCheckState);
+        return newCheckState;
+    }
+
+    public static void setAllToCheckedState(ListView lv, boolean check) {
+        // no-op if ListView null
+        if (lv == null) return;
+
+        for (int x = 0; x < lv.getCount(); x++) {
+            lv.setItemChecked(x, check);
+        }
+    }
+
+    // Function to toggle button label
+    public static void toggleButtonLabel(Button mToggleButton, ListView lv) {
+        if (lv.getCheckedItemCount() != lv.getCount()) {
+            mToggleButton.setText(R.string.select_all);
+        } else {
+            mToggleButton.setText(R.string.clear_all);
+        }
+    }
+
+
+    @Override
+    public void setUserVisibleHint(boolean isVisibleToUser) {
+        if (!isVisibleToUser) {
+            // close the drawer if open
+            if (mDrawerLayout != null && mDrawerLayout.isDrawerOpen(Gravity.END))
+                mDrawerLayout.closeDrawer(Gravity.END);
```

Comment:
I am in favour of brackets even if there is only one line. It doesn't look good now.